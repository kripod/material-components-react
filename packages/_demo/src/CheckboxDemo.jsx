import React, { PropTypes, PureComponent } from 'react';
import Checkbox from '../../checkbox';

export default class CheckboxDemo extends PureComponent {
  static propTypes = {
    nativeElementId: PropTypes.string,
  };

  static defaultProps = {
    nativeElementId: 'demo-checkbox',
  };

  state = {
    indeterminate: false,
    disabled: false,
  };

  checkboxElement = {};

  render() {
    return (
      <div>
        <div className="mdc-form-field">
          <Checkbox
            ref={(element) => { this.checkboxElement = element; }}
            id={this.props.nativeElementId}
            indeterminate={this.state.indeterminate}
            disabled={this.state.disabled}
          />
          <label htmlFor={this.props.nativeElementId}>This is my checkbox</label>
        </div>

        <button onClick={() => this.setState({ indeterminate: !this.state.indeterminate })}>
          Make Indeterminate
        </button>

        <button onClick={() => this.setState({ disabled: !this.state.disabled })}>
          Toggle Disabled
        </button>
      </div>
    );
  }
}
