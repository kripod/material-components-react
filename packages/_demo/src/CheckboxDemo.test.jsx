import React from 'react';
import renderer from 'react-test-renderer';
import CheckboxDemo from './CheckboxDemo';

function createNodeMock() {
  return {
    addEventListener: () => {},
  };
}

it('renders correctly', () => {
  const tree = renderer.create(<CheckboxDemo />, { createNodeMock }).toJSON();
  expect(tree).toMatchSnapshot();
});
