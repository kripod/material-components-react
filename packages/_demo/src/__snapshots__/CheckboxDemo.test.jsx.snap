exports[`test renders correctly 1`] = `
<div>
  <div
    className="mdc-form-field">
    <div
      checked={false}
      className="mdc-checkbox  "
      id="demo-checkbox">
      <input
        checked={false}
        className="mdc-checkbox__native-control"
        disabled={false}
        onChange={[Function]}
        type="checkbox" />
      <div
        className="mdc-checkbox__background">
        <svg
          className="mdc-checkbox__checkmark"
          version="1.1"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg">
          <path
            className="mdc-checkbox__checkmark__path"
            d="M1.73,12.91 8.1,19.28 22.79,4.59"
            fill="none"
            stroke="white" />
        </svg>
        <div
          className="mdc-checkbox__mixedmark" />
      </div>
    </div>
    <label
      htmlFor="demo-checkbox">
      This is my checkbox
    </label>
  </div>
  <button
    onClick={[Function]}>
    Make Indeterminate
  </button>
  <button
    onClick={[Function]}>
    Toggle Disabled
  </button>
</div>
`;
